tests/receivers/jmxreceiver/jmxreceiver_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build integration
+
+package tests
+
+import (
+	"os"
+	"path"
+	"testing"
+
+	"github.com/signalfx/splunk-otel-collector/tests/testutils"
+)
+
+func TestJMXReceiverProvidesAllJVMMetrics(t *testing.T) {
+	// JMX metrics are missing when running on arm so the test only checks a subset
+	// of what would be received otherwise.
+	expected_metrics_file := "all.yaml"
+	if testutils.CollectorImageIsForArm(t) {
+		expected_metrics_file = "arm64.yaml"
+	}
+
+	containers := []testutils.Container{
+		testutils.NewContainer().WithContext(
+			path.Join(".", "testdata", "server"),
+		).WithExposedPorts("7199:7199").WithName("jmx").WillWaitForPorts("7199"),
+	}
+
+	testutils.AssertAllMetricsReceived(t, expected_metrics_file,
+		"all_metrics_config.yaml", containers,
+		[]testutils.CollectorBuilder{
+			func(collector testutils.Collector) testutils.Collector {
+				// JMX requires a local directory that can be written to, so we must mount a local dir
+				// that the collector has write access to.
```

Comment:
Doesn't this mean the jmx receiver won't work without this having been set in the collector image for end users? I think we should create an internal tmpdir that `TMPDIR` points to by default and document that this is required for `--tmpfs` if the collector is run w/ a read-only fs.